src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -1317,7 +1307,7 @@ public LdapUserDetails loadUserByUsername(String username) throws UsernameNotFou
                         }
                     }
                 }
-                LdapUserDetails ldapUser = ldapSearch.searchForUser(username);
+                LdapUserDetails ldapUser = ldapSearch.searchForUser(username); // TODO mismatch between DirContextOperations+String vs. LdapUserDetails
```

Comment:
Here is one of several places where I could not figure out how to get things to even compile. If I change the type of the variable to `DirContextOperations` then some errors go away (calls to `getGrantedAuthorities` and `getAttributes`) but many others appear.

Reply:
I've been looking into this. `LdapUserDetailsManager` seems to be a replacement for the whole method.

However, we would need a custom `UserDetailsContextMapper` in order to map `LDAPConfiguration#displayNameAttributeName` and `LDAPConfiguration#mailAddressAttributeName`

Reply:
Great, I will look into that.

Reply:
Looked at it, but that seemed to be a way bigger refactoring, and I did not see how to fit it into the plugin’s current logic. For now, went with a simpler patch that at least fixes the types here.